<launch>


  <param name="robot_description"
    textfile="$(find magnetic_controller)/robots/simple_robot.urdf" />

  <node pkg="iai_naive_kinematics_sim" type="velocity_resolved_sim" 
        name="velocity_resolved_sim" output="screen">
    <rosparam param="controlled_joints" file="$(find magnetic_controller)/robots/controlled_joints.yaml" />
    <rosparam param="start_config"
      file="$(find magnetic_controller)/robots/start_config.yaml" />
    <param name="sim_frequency" value="100.0"/>
    <param name="watchdog_period" value="0.2"/>
  </node>

  <node pkg="robot_state_publisher" type="robot_state_publisher"
        name="robot_state_publisher" />

  <node pkg="tf" type="static_transform_publisher" name="fake_odometry"
        output="screen" args="0 0 0 0 0 0 1 odom_combined link0 100" />

  <node pkg="tf" type="static_transform_publisher" name="cube1"
        output="screen" args="2 0 0 0 0 0 1 odom_combined cube1 100" />

  <node pkg="tf" type="static_transform_publisher" name="cylinder1"
        output="screen" args="2 1 0 0 0 0 1 odom_combined cylinder1 100" />

  <node pkg="tf" type="static_transform_publisher" name="cylinder2" 
        output="screen" args="2 -1 0 0 0 0 1 odom_combined cylinder2 100" />

</launch>